hdf/src/hcompri.c

```diff
@@ -448,7 +448,7 @@ HRPcloseAID(accrec_t *access_rec)
        If no more references to that, free the record */
 
     if (--(info->attached) == 0) {
-        HDfree(info);
+        free(info);
```

Comment:
We should stop using local alias for access_rec->special_info pointer - it's just a bad smell waiting for a problem. 
This function could just be:
`if (--(access_rec->special_info->attached) == 0)
  free(access_rec->special_info)
return SUCCEED`

Reply:
In this particular case, it doesn't matter, but in general, a function needs to know which special info it is going to access: chunkinfo_t, compinfo_t, crinfo_t, extinfo_t,... and it's simpler to use local alias.
Having access_rec->special_info everywhere makes the code lengthy and hard to read, in my opinion.

Reply:
But having a generic alias may lead to misunderstandings that may lead to weird errors if not properly handled.
You know some github user trys to help and creates a mess because of not fully reading code.
I like the idea of just setting the access_rec->special_info upon success.